File: BeautySaloon/BeautySaloon.Angular/src/app/shared/models/service.ts
import { Master } from "./master";
import { Reservation } from "./resrvation/reservation";
import { ServiceCategory } from "./service-category";

export interface Service {
  id: number | null;
  name: string | null;
  price: number | null;
  categoryId: number | null;
  category: ServiceCategory | null;
  duration: string | null;  // You can use string or number (in minutes) depending on your needs
  reservations: Reservation[] | null;
  masters: Master[] | null;
}

File: BeautySaloon/BeautySaloon.API/Entities/BeautySaloonContextEntities/Service.cs
namespace BeautySaloon.API.Entities.BeautySaloonContextEntities;

public class Service
{
    public int? Id { get; set; }
    public string? Name { get; set; }
    public decimal? Price { get; set; }
    public int? CategoryId { get; set; }
    public ServiceCategory? Category { get; set; }
    public TimeSpan? Duration { get; set; } = TimeSpan.FromHours(1);
    public List<Reservation>? Reservations { get; set; }
    public List<Master>? Masters { get; set; }
}
